hbase-common/src/main/java/org/apache/hadoop/hbase/types/OrderedBlob.java

```diff
@@ -32,7 +32,7 @@
   public static final OrderedBlob ASCENDING = new OrderedBlob(Order.ASCENDING);
   public static final OrderedBlob DESCENDING = new OrderedBlob(Order.DESCENDING);
 
-  protected OrderedBlob(Order order) {
+  public OrderedBlob(Order order) {
```

Comment:
I think this is intentional? User can use the above ASCENDING and DESCENDING directly?

Reply:
I would guess so. @ndimiduk probably can answer this question better.

Reply:
So that's precisely it. See my last comment on JIRA. The original design of this part of the API was to try to prevent callers from needless allocations. That couldn't be done uniformly, though, as some of these implementations require explicit construction parameters. I myself filed a ticket wondering why I had left some constructors private. I believe the uniformity of public constructors across all concrete types is more consistent that having static constants in most, but not all, places.

This is definitely a subjective area of the API design and I'd love to get other opinions :)

Reply:
To make it consistent, how about making the constructors public for consistency reasons and deprecate the static constants (and remove them in another release) to make it more consistent?

Reply:
If you think deprecating the static constants is appropriate, then I have no objection. There's always a balance between minimizing our supported surface area and downstream convenience. I've not followed things very closely as of late, but it seems we're on a pretty strong kick to deprecate/drop old interfaces.

Reply:
I think that we should discuss this on the dev list. @Apache9 Is your question answered? What do you think? Should we leave it as is or should we deprecate the static constants?